### DISCUSSION TITLE
Mulitplexing with an IC4 and other questions.

### DISCUSSION THREAD

**Discussion by DadLovesBluey** [doudar/SmartSpin2k#234] [Q&A]
Hi All! I recently found out about this amazing project.  I haven't built a full unit, but I did get a dev board (a little different brand than the recommended as it was out of stock on Amazon). I have the SmartSpin2k firmware/software running and I'm able to connect my heart rate monitor to it.  My plan was to initially connect my Schwinn IC4 to the smartspin2k dev board and use it to multiplex the bike and heart sensors to two different applications simultaneously (if that is supported?).  Longterm, I'd like to build out the rest of the project so that I could have the resistance adjustment even if multiplexing isn't supported.  I apologize in advance for all of the newb questions, and if this isn't the proper place for them, please let me know. 

Questions:
1) Is my plan to multiplex the two sensors to two different apps via the smartspin2k build supported? If it is, is there a limit on how many applications can simultaneously access the sensors connected to the smartspin2k?

2) Should the IC4 show up as a "Power Meter" choice in the application after I do a bluetooth scan?  It doesn't seem to show up there, but if I go into the debug, I can see that it is still somehow still registering the cadence from the bike

3) I know that the IC4 doesn't typically provide a true power metric via bluetooth, but I know other apps (kinetic, QZ, mpaceline, etc.) are able to estimate IC4 power based on an algorithm involving cadence/resistance.  Is that something that smartspin2k does or could do in the future? or is using the heartrate estimate of power the only (or best) choice at this time?

4) Has anyone used the Peloton Digital App and connected to the smartspin2k as the cadence sensor?

**Comment by chinasaur** [comment]
Welcome! I also have an IC4 (well C6 but same thing). I haven't gotten into the software side yet other than building a recent checkout though. I've been working on breaking out the up/down console buttons to use as shifters, as well as potentially breaking out the resistance sensor I2C to get faster resistance feedback than is available over bluetooth. I think I almost have this working; will send update when I do.

Others can say more on your software questions; looking forward to hearing the answers and how your build goes.

**Comment by doudar** [comment]
> Hi All! I recently found out about this amazing project. I haven't built a full unit, but I did get a dev board (a little different brand than the recommended as it was out of stock on Amazon). I have the SmartSpin2k firmware/software running and I'm able to connect my heart rate monitor to it. My plan was to initially connect my Schwinn IC4 to the smartspin2k dev board and use it to multiplex the bike and heart sensors to two different applications simultaneously (if that is supported?). Longterm, I'd like to build out the rest of the project so that I could have the resistance adjustment even if multiplexing isn't supported. I apologize in advance for all of the newb questions, and if this isn't the proper place for them, please let me know.
> 
> Questions:
> 
> 1. Is my plan to multiplex the two sensors to two different apps via the smartspin2k build supported? If it is, is there a limit on how many applications can simultaneously access the sensors connected to the smartspin2k?
> 2. Should the IC4 show up as a "Power Meter" choice in the application after I do a bluetooth scan?  It doesn't seem to show up there, but if I go into the debug, I can see that it is still somehow still registering the cadence from the bike
> 3. I know that the IC4 doesn't typically provide a true power metric via bluetooth, but I know other apps (kinetic, QZ, mpaceline, etc.) are able to estimate IC4 power based on an algorithm involving cadence/resistance.  Is that something that smartspin2k does or could do in the future? or is using the heartrate estimate of power the only (or best) choice at this time?
> 4. Has anyone used the Peloton Digital App and connected to the smartspin2k as the cadence sensor?

Hi there! 

Yes, you should be able to connect SS2K to 2 different devices (clients) simultaneously. The only beware is that currently both connected clients can set parameters (they can attempt to control) the SS2K at the same time so you need to make sure that one is only reading the information (i.e. in Zwift, set as a power meter only, not a smart trainer). 

The IC4 should show up on the webpage as a named power meter. 

As far as I know, the IC4 does actually output a power estimate but it's known to be inaccurate in a non linear way. (like under reported at <200w and over reported at >300w . We've got a configurable power correction factor to accommodate for this but it's an (linear) correction. Someday maybe we can implement a multipoint correction.
